test/ably/realtimechannel_test.py

```diff
@@ -38,3 +42,152 @@ async def test_channel_detach(self):
         await channel.detach()
         assert channel.state == ChannelState.DETACHED
         await ably.close()
+
+    # RTL7b
+    async def test_subscribe(self):
+        ably = await RestSetup.get_ably_realtime()
+        await ably.connect()
+        channel = ably.channels.get('my_channel')
+        await channel.attach()
+        listener = Mock()
+        await channel.subscribe('event', listener)
+
+        # publish a message using rest client
+        rest = await RestSetup.get_ably_rest()
+        rest_channel = rest.channels.get('my_channel')
+        await rest_channel.publish('event', 'data')
+        await asyncio.sleep(0.5)
```

Comment:
Yeah that's a fair point, I've refactored it a bit now and removed most of the sleeps. For testing `unsubscribe` there isn't a better alternative unfortunately... if you're expecting the listener to not be called then there isn't an event to wait for - you just have to give it some time so that if a message is received and the listener is called it will fail (well, unless we set up an interface for the tests to listen to individual protocol messages but that would be a faff and probably not worth the effort)